components/org.wso2.carbon.identity.api.server.permission.management/org.wso2.carbon.identity.api.server.permission.management.v1/src/main/java/org/wso2/carbon/identity/api/server/permission/management/v1/impl/PermissionManagementApiServiceImpl.java

```diff
@@ -0,0 +1,41 @@
+/*
+* Copyright (c) 2019, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+*
+* Licensed under the Apache License, Version 2.0 (the "License");
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+* http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+package org.wso2.carbon.identity.api.server.permission.management.v1.impl;
+
+import org.json.JSONArray;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.wso2.carbon.identity.api.server.permission.management.v1.PermissionManagementApiService;
+import org.wso2.carbon.identity.api.server.permission.management.v1.core.PermissionManagementService;
+
+import java.util.Arrays;
+import javax.ws.rs.core.Response;
+
+/**
+ * The Permission Management API implementation class.
+ */
+public class PermissionManagementApiServiceImpl implements PermissionManagementApiService {
+
+    @Autowired
+    private PermissionManagementService permissionManagementService;
+
+    @Override
+    public Response permissionManagementPermissionsGet() {
+
+        JSONArray permissions = new JSONArray(Arrays.asList(permissionManagementService.getAllPermissions()));
```

Comment:
There is no need to convert this to json objects. If the PermissionObject list is returned, it will be properly serialized to a json array.